plugins/inputs/slurm/README.md

```diff
@@ -0,0 +1,227 @@
+# SLURM Input Plugin
+
+This plugin gather diag, jobs, nodes, partitions and reservation metrics by
+leveraging SLURM's REST API as provided by the `slurmrestd` daemon.
+
+This plugin targets the `openapi/v0.0.38` OpenAPI plugin as defined in SLURM's
+documentation. That particular plugin should be configured when starting the
+`slurmrestd` daemon up. For more information, be sure to check SLURM's
+documentation [here][SLURM Doc].
```

Comment:
Hmm do you have any idea how often this changes? I want to be sure we aren't signing up for something that is changing too often. The website makes it clear that there is no guarantee of compatibility between versions. In general, I would want to have more confidence in an API's stability before adding a plugin.

Reply:
TL;DR: Each OpenAPI plugin version is supported for roughly 4 years or, at least, that's what I gather from the documentation 😅 

The documentation states that  with each major (i.e. yearly) release they mark the oldest plugin as deprecated so that it's removed on the next major release. For instance, I tested the API client against `openapi/v0.0.38` which ships with SLURM `22.05.9` (i.e. a release from 2022) and that plugin is still present in SLURM's latest release which is `24.05`. Given their usual workflow, `openapi/v0.0.38` will still be around for SLURM `25.x.y` and will be removed on SLURM `26.x.y`.

However, there's nothing better than reading the original documentation to be sure which you can find [here](https://slurm.schedmd.com/archive/slurm-22.05.9/rest.html#plugins) for `22.05.9` and [here](https://slurm.schedmd.com/rest.html#plugins) for `24.05`. You can also access a version archive [here](https://slurm.schedmd.com/archive/).

I would be more than happy to develop the API client for `openapi/v0.0.39` (which is the latest) too and then add either a discovery mechanism on `func (s *Slurm) Init() error` or a configuration option to choose among versions.

If you need any additional details please do let me know.

Reply:
Thanks for the details, let me look through this and get back to you.

Reply:
Hi! It looks like this is the last remaining thread. Is there any more information I can provide on the topic? 🤔

Reply:
Given the versions I'm seeing in Debian/Ubuntu/Fedora this looks fine. Thanks agian for the background.